File: backend/src/modules/plano_modalidade/entity/PlanoModalidade.js
class PlanoModalidade {
  constructor({ plano_id_plano, modalidade_id_modalidade }) {
    this.plano_id_plano = plano_id_plano;
    this.modalidade_id_modalidade = modalidade_id_modalidade;
  }
  validate() {
    const errors = [];
    if (!this.plano_id_plano) errors.push('Plano é obrigatório');
    if (!this.modalidade_id_modalidade) errors.push('Modalidade é obrigatória');
    return errors;
  }
}
module.exports = PlanoModalidade;

File: backend/src/main/java/com/vitalativa/entity/PlanoModalidade.java
package com.vitalativa.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;

@Entity
@Table(name = "plano_modalidade")
public class PlanoModalidade {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Integer id;
    
    @NotNull(message = "Plano é obrigatório")
    @Column(name = "plano_id_plano")
    private Integer planoIdPlano;
    
    @NotNull(message = "Modalidade é obrigatória")
    @Column(name = "modalidade_id_modalidade")
    private Integer modalidadeIdModalidade;
    
    public PlanoModalidade() {}
    
    public Integer getId() { return id; }
    public void setId(Integer id) { this.id = id; }
    public Integer getPlanoIdPlano() { return planoIdPlano; }
    public void setPlanoIdPlano(Integer planoIdPlano) { this.planoIdPlano = planoIdPlano; }
    public Integer getModalidadeIdModalidade() { return modalidadeIdModalidade; }
    public void setModalidadeIdModalidade(Integer modalidadeIdModalidade) { this.modalidadeIdModalidade = modalidadeIdModalidade; }
}